sites/docs/src/content/docs/tutorials/tests_and_test_data/components/03_testing_modules.md

```diff
@@ -0,0 +1,669 @@
+- `process` - **Process name**: The specific process to test (must match the process name in `main.nf`)
+- `test("...")` - **Individual test case**: Each test block defines one scenario to test
+- `when { }` - **Test setup**: Where you define the input data for the process
+- `then { }` - **Assertions**: Where you verify the process behaved as expected
+
+**Inside the test components:**
+
+Now that we understand the overall structure, let's look at the specific elements within the test:
+
+- `[ id:'test' ]` - **meta map**: Contains sample metadata (ID, conditions, etc.)
+- `file('test_file.txt')` - **input file**: The file to process
+- `process.success` - **success check**: Ensures the process completed without errors
+- `snapshot(sanitizeOutput(process.out)).match()` - **output validation**: Compares all outputs to stored snapshot
+- `sanitizeOutput()` - **snapshot sanitization**: Cleans process and workflow outputs by removing numbered keys, making snapshots more human-readable
+
+### Step 3: Run the test
+
+Once you've written the test, you can generate the initial snapshot by running the test.
+
+```bash
+cd path/to/modules
+nf-core modules test cat --profile docker
+```
+
+**What happens:**
+
+1. nf-test creates a temporary work directory `.nf-test/`
+2. Runs the CAT process with your test inputs
+3. **First run**: Generates `tests/main.nf.test.snap` with output checksums
+4. **Second run**: Validates outputs match the generated snapshot
+5. Reports PASS/FAIL
+
+### Step 4: Examine the generated snapshot
+
+After running, check `tests/main.nf.test.snap`:
+
+```json
+{
+  "simple file concatenation": {
+    "content": [
+      {
+        "0": [
+          // First output channel (file_out)
+          [{ "id": "test" }, "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"]
+        ],
+        "1": [
+          // Second output channel (versions)
+          "versions.yml:md5,c50aa59475ab20752531545a585f8f2d"
+        ]
+      }
+    ]
+  }
+}
+```
+
+You can see the the outputs of the module, with an associated `md5sum` which concisely describes the exact contents of the file.
+
+Now you understand the full test cycle! The snapshot ensures your module produces consistent, expected outputs every time.
+
+Any time you change the module - you can run the test again to check to see you did not unexpectedly change an output that should not have been changed.
+
+## Essential assertions
+
+Tests use assertions to verify the expected output of the process specified in the `then` block.
+
+You can specify multiple assertions to be evaluated together in a single test by specifying them within an `assertAll` block.
+
+Nextflow process output channels that lack explicit names (i.e., when no [`meta` map](https://nf-co.re/docs/contributing/components/meta_map) is present) can be addressed using square brackets and the corresponding index, for example `process.out[0]` for the first channel and `process.out[0][1]` for the second element of the first channel.
```

Comment:
Uh this is confusing now, the `meta` map doesn't influence the explicit names. Maybe rephrase slightly... maybe more 'channels with multple objects in a tuple) or something.